runbot_travis2docker/models/runbot_branch.py

```diff
@@ -3,20 +3,86 @@
+from datetime import datetime
+
 from openerp import fields, models, api
 
 
 class RunbotBranch(models.Model):
     _inherit = "runbot.branch"
 
     uses_weblate = fields.Boolean(help='Synchronize with Weblate')
+    updated_weblate = fields.Datetime(help='Last update of weblate')
+    name_weblate = fields.Char(compute='_compute_name_weblate', store=True)
+
+    @api.multi
+    @api.depends('repo_id.name', 'branch_name', 'uses_weblate')
+    def _compute_name_weblate(self):
+        for branch in self:
+            name = branch.repo_id.name.replace(':', '/')
+            name = re.sub('.+@', '', name)
+            name = re.sub('.git$', '', name)
+            name = re.sub('^https://', '', name)
+            name = re.sub('^http://', '', name)
+            match = re.search(
+                r'(?P<host>[^/]+)/(?P<owner>[^/]+)/(?P<repo>[^/]+)', name)
+            if match:
+                name = ("%(host)s:%(owner)s/%(repo)s (%(branch)s)" %
+                        dict(match.groupdict(), branch=branch['branch_name']))
+            branch.name_weblate = name
 
     @api.model
     def cron_weblate(self):
         for branch in self.search([('uses_weblate', '=', True)]):
-            self.env['runbot.build'].create({'branch_id': branch.id,
-                                             'name': 'HEAD',
-                                             'uses_weblate': True})
+            if (not branch.repo_id.weblate_token or
+                    not branch.repo_id.weblate_url):
+                continue
+            current_date = (datetime.strptime(branch.updated_weblate,
+                                              '%Y-%m-%d %H:%M:%S')
+                            if branch.updated_weblate
+                            else None)
+            new_date = None
+            url = branch.repo_id.weblate_url
+            session = requests.Session()
+            session.headers.update({
+                'Accept': 'application/json',
+                'User-Agent': 'runbot_travis2docker',
+                'Authorization': 'Token %s' % branch.repo_id.weblate_token
+            })
+            projects = session.get(url + '/projects/').json()
+            for project in projects['results']:
+                components = session.get('%s/projects/%s/components'
+                                         % (url, project['slug'])).json()
+                for component in components['results']:
+                    if component['branch'] != branch['branch_name']:
+                        continue
+                    if project['name'] != branch.name_weblate:
+                        continue
+                    changes = session.get('%s/components/%s/%s/changes/'
+                                          % (url, project['slug'],
+                                             component['slug'])).json()
+                    if not changes['results']:
+                        continue
+                    change = iter(changes['results']).next()
```

Comment:
I don't understand `iter` use because if we have big list you will have a extra process just to get the first item.

What about use the first with index?
`change = changes['results'][0]`